kernel/src/main/java/org/kframework/unparser/WantsPrettyPrinter.java

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2019 K Team. All Rights Reserved.
+package org.kframework.unparser;
+
+/**
+ * To be implemented by instances of {@code Rewriter} that need an object of type PrettyPrinter.
+ *
+ * @author Denis Bogdanas
+ * Created on 08-Feb-19.
+ * @see org.kframework.rewriter.Rewriter
+ */
+public interface WantsPrettyPrinter {
```

Comment:
I'm reasonably certain you don't need this interface You can just construct an instance of PrettyPrinter inside SymbolicRewriterGlue.

Reply:
I scratched my head enough to come with least invasive solution. I cannot construct PrettyPrinter in SymbolicRewriterGlue, because I don't have access to Definition and Module there. Alternative would be to have a `WantsDefinitionAndModule` instead. Doesn't look better.

Reply:
I'm super confused. A SymbolicRewriterGlue is already specific to a specific definition and module, it would be utterly trivial to ensure you had access to the appropriate information in that class, assuming it's not already there.

Reply:
Please look at my reply below. Looking at the whole code in Intellij also would help. To pass the right information to SymbolicRewriterGlue, one way or another, I have to alter the interfaces that create it. This will affect all backends.